cpp/src/Ice/Protocol.cpp

```diff
@@ -70,8 +71,9 @@ namespace IceInternal
         byte{0} // Message size (placeholder)
     };
 
-    void stringToMajorMinor(const std::string& str, uint8_t& major, uint8_t& minor)
+    void stringToMajorMinor(string_view s, uint8_t& major, uint8_t& minor)
     {
+        string str{s};
         std::string::size_type pos = str.find_first_of(".");
         if (pos == std::string::npos)
```

Comment:
What is the point of passing `string_view` to this method, if we construct a `string` right away?

Reply:
It's because the public API should use string_view.

The implementation can keep using strings, that's fine.